### PR TITLE
Show parent functionality

### PR DESCRIPTION
**Pull request by orozcodiaz**
### Description
This feature adds a button "Show Parent" to the top of simple products in the admin panel.
It helps to manage products. With this feature, you can save time and never go back to the grid, search parent product etc.

### Manual testing scenarios
1. Add simple product
2. Add virtual/downloadable product
3. Add configurable product
4. Try to go back by clicking button from every scenario (if button exists)

### Visual

Video of how it works: http://recordit.co/1AKbNlgeKD

screenshot 2018-08-13 19 27 14

### CODE REVIEW

**Comment by magento-cicd2** [comment]
[CLA assistant check](https://cla.dev.magento.com/magento/magento2?pullRequest=17581) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla.dev.magento.com/magento/magento2?pullRequest=17581) before we can accept your contribution.<br/>**4** out of **10** committers have signed the CLA.<br/><br/>:white_check_mark: magento-engcom-team<br/>:white_check_mark: DanielRuf<br/>:white_check_mark: sidolov<br/>:white_check_mark: orlangur<br/>:x: o-iegorov<br/>:x: ameysar<br/>:x: svitja<br/>:x: magento-cicd2<br/>:x: olysenko<br/>:x: DianaRusin

**Comment by magento-engcom-team** [comment]
Hi @oroskodias. Thank you for your contribution 
Here is some useful tips how you can test your changes using Magento test environment.
Add the comment under your pull request to deploy test or vanilla Magento instance:
- `@magento-engcom-team give me test instance` - deploy test instance based on PR changes
- `@magento-engcom-team give me {$VERSION} instance` - deploy vanilla Magento instance

For more details, please, review the [Magento Contributor Assistant](https://github.com/magento/magento2/wiki/Magento-Contributor-Assistant) documentation

File: app/code/Magento/Catalog/Block/Adminhtml/Product/Edit/Button/ShowParent.php [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+use Magento\ConfigurableProduct\Model\ResourceModel\Product\Type\Configurable;
+use Magento\Framework\View\Element\UiComponent\Context;
+use Magento\Framework\Registry;
+
+class ShowParent extends Generic
+{
+    /**
+     * @var Configurable
+     */
+    protected $configurableProduct;
```
Comment (osrecio) [inline app/code/Magento/Catalog/Block/Adminhtml/Product/Edit/Button/ShowParent.php:20]: Shoud be private

**Comment by osrecio** [comment]
Hi @oroskodias,

Thanks for your contribution. I have a question:

What happens if a simple product has more than 1 parent?. 1 simple in 2 or more configurable.

Check the conflict file, please

**Review by osrecio — CHANGES_REQUESTED** [review]

File: app/code/Magento/Catalog/Block/Adminhtml/Product/Edit/Button/ShowParent.php [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    public function getButtonData()
+    {
+        $product = $this->getProduct();
+        $configurableProduct = $this->configurableProduct->getParentIdsByChild($product->getId());
+        
+        if(!isset($configurableProduct[0]) || empty($configurableProduct[0])){
+            return parent::getButtonData();
+        }
+
+        if ($product->getTypeId() == Type::TYPE_SIMPLE && $product->getVisibility() == Visibility::VISIBILITY_NOT_VISIBLE) {
```
Comment (osrecio) [inline app/code/Magento/Catalog/Block/Adminhtml/Product/Edit/Button/ShowParent.php:50]: Why should be Not visible to show a button? Perhaps you have simple products that you want show in frontend and show button parent

**Review by osrecio — CHANGES_REQUESTED** [review]

File: app/code/Magento/Catalog/Block/Adminhtml/Product/Edit/Button/ShowParent.php [hunk 3]
```diff
@@ -0,0 +1,57 @@
+<?php
+
```
Comment (osrecio) [inline app/code/Magento/Catalog/Block/Adminhtml/Product/Edit/Button/ShowParent.php]: missed Magento CopyRight

```
/**
 * Copyright © Magento, Inc. All rights reserved.
 * See COPYING.txt for license details.
 */
```

**Review by osrecio — CHANGES_REQUESTED** [review]

File: app/code/Magento/Catalog/Block/Adminhtml/Product/Edit/Button/ShowParent.php (hunk 2 above)
Comment (osrecio) [reply app/code/Magento/Catalog/Block/Adminhtml/Product/Edit/Button/ShowParent.php:50]: Better with `===`

**Review by osrecio — CHANGES_REQUESTED** [review]

**Comment by orlangur** [comment]
@oroskodias, unfortunately, we can't accept such Pull Requests to 2.2 release line, new features should be delivered to 2.3 release line. For more information, please, refer to [Magento Code Contributions Guide](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#rules).

File: app/code/Magento/Catalog/Block/Adminhtml/Product/Edit/Button/ShowParent.php [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+namespace Magento\Catalog\Block\Adminhtml\Product\Edit\Button;
+
+use Magento\Catalog\Model\Product\Type;
+use Magento\Catalog\Model\Product\Visibility;
+use Magento\ConfigurableProduct\Model\ResourceModel\Product\Type\Configurable;
```
Comment (orlangur) [inline app/code/Magento/Catalog/Block/Adminhtml/Product/Edit/Button/ShowParent.php:11]: Configurable product cannot be referred in Catalog module.

Also, all other composite product types which may contain simple product must be taken into consideration: grouped, bundle.

**Review by orlangur — CHANGES_REQUESTED** [review]

**Pull request closed without merging**